test/functional/tools/parameters/gx_int_required.xml

```diff
@@ -0,0 +1,11 @@
+<tool id="gx_int_required_via_empty_string" name="gx_int_required_via_empty_string" version="1.0.0">
```

Comment:
This id `gx_int_required_via_empty_string` seems wrong. Its used in another test tool - below. A test section is missing.

Reply:
Good catch - thank you! I think the missing tests are fine - I just want to test parameter validation against the parameter set. I've been adding so many test tools - which I think are great in terms of catching corner cases in various ways and being able to inspect the stack in isolation - but if I added tests for all of them might slow down the test suite substantially and be really redundant.

I'll rebase with the id fixed.

Reply:
The matching tests for this behavior are found in integer_default.xml added by Nicola last year.